### PR TITLE
Update vulture container image to run as non-root

### PR DESCRIPTION
**Pull request by pharaujo**
**What this PR does**:

Follow-up to #2265 which missed tempo-vulture

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pharaujo** [comment]
@zalegrala would you mind having a look?

File: cmd/tempo-vulture/Dockerfile [hunk 1]
```diff
@@ -2,4 +2,13 @@ FROM alpine:3.20 as certs
 RUN apk --update add ca-certificates
 ARG TARGETARCH
 COPY bin/linux/tempo-vulture-${TARGETARCH} /tempo-vulture
+
+RUN addgroup -g 10001 -S tempo && \
+    adduser -u 10001 -S tempo -G tempo
+
+RUN mkdir -p /var/tempo && \
```
Comment (zalegrala) [inline cmd/tempo-vulture/Dockerfile:9]: Vulture doesn't write data locally, and so the mkdir and associated chown aren't required.

Comment (pharaujo) [reply cmd/tempo-vulture/Dockerfile]: Thanks, fixed.

**Review by zalegrala — APPROVED** [review]
This looks good to me.  Did you have a chance to build an image and test it?

**Comment by pharaujo** [comment]
> Did you have a chance to build an image and test it?

Yes, I build it and ran in Kubernetes with `runAsNonRoot: true` securityContext, so everything looks good.

**Comment by zalegrala** [comment]
Hmm, the lint is failing and github isn't letting me retry.  Could you try a rebase to force CI to try again?

**Pull request merged**